### DISCUSSION TITLE
Support for the Delete feature in version 7

### DISCUSSION THREAD

**Discussion by siddhishinde0723** [Sunbird-Knowlg/Community#181] [General Discussions]
Hi Team, 

We would like to know how the delete functionality added in release 7 works. 
Where can we obtain the synopsis for the same?

**Comment by pallakartheekreddy** [comment]
@siddhishinde0723 Please share more details?

**Reply by siddhishinde0723** [reply]
@pallakartheekreddy 
An overview about the delete functionality.
What all data it can delete?
Will it delete data form neo4j, kasandra , elastic search ?

**Reply by krgauraw** [reply]
Hi @siddhishinde0723 When you delete an user in sunbird Ed, Below will happen at Knowlg Level:

- Knowg uses a flink job "user-pii-data-updater" from inQuiry which will process the PII (Personal Identifiable Information) Cleanup.
- The job works for all object type which is configured under "target_object_types". e.g: Content, Collection, Asset
- The field which holds PII info, can be configured in schema for each object type. e.g: Content, Collection. By default, creator field is configured as target PII metdata.

Sample of PII config: https://github.com/Sunbird-Knowlg/knowledge-platform/blob/2ff1a7e5c2d7a02443f8ad3ac7ffe10138401b04/schemas/asset/1.0/config.json#L104

- In above config, lookup key is "createdBy" and target field is "creator"

- The flink job read the event generated for deleted user and do as below:
   - search in all object types (configured in traget_object_types) with configured lookup key in each object schema for deleted user.
   - cleanup the graph data (neo4j) for target field configured in each object schema. Elasticsearch records would be automatically handled by knowlg search-indexer job.
   - trigger an email notification to org admin with affected do_ids.
   
 Note: the flink job won't make any changes to hierarchal structure (in case of collection). Basically no data update in cassandra.
 So if you have any Live status asset, you need to re-publish them in order to reflect the changes. If you have any  collection in draft/review status which may hold deleted user information even after pii data cleanup, the latest changes to PII will reflect when it goes for publish.
 
 For more information on the job, please use below link: 
 https://inquiry.sunbird.org/use/release-notes/inquiry-release-v7.0.0-latest#question-and-question-set-service
 Hope this helps!
 
 cc: @pallakartheekreddy

**Reply by pallakartheekreddy** [reply]
@siddhishinde0723 As we haven't received a response from you, we will proceed to close this thread. If you have any further questions or require assistance in the future, please don't hesitate to start a new thread. Thank you.
